testing_support/BaseIntegrationTest.py

```diff
@@ -28,6 +28,11 @@ def tearDown(self) -> None:
     def remove_destination(self):
```

Comment:
did you test that this implementation also works for multiple nested files, e.g. /test_project/subfolder/subsubfolder/some_file.dp.cpp ? It should work for any type of nesting (any amount of layers). Back when we used the script, Zhongyuan had an elegant solution here where the function calls itself if subfolders are identified.